apps/web/src/components/SearchBar/SearchBar.tsx

```diff
@@ -0,0 +1,44 @@
+import * as React from 'react';
+import { useState } from 'react';
+import './SearchBar.scss';
+
+export interface Props {
+  /** What happens when we use a searchstring to search for results? */
+  handleSearch: (searchString: string) => void;
+  /**  An optional custom test ID for the component.  */
+  customTestId?: string;
+}
+
+// need to add jsdoc comment
+const SearchBar = ({
+  handleSearch,
+  customTestId = 'searchbar',
+}: Props): React.ReactElement => {
+  /** Tracks the string value within input */
+  const [searchString, setSearchString] = useState<string>('');
+
+  /**
+   * Handles 'Enter' key press & invokes `handleSearch` func
+   *
+   * @param {React.KeyboardEvent} event - The keyboard event that triggered the function.
+   */
+  const handleKeyDown = (event: React.KeyboardEvent): void => {
+    if (event.key === 'Enter') {
+      handleSearch(searchString);
+    }
+  };
+
+  return (
+    <div className="searchbar-container" data-testid="searchbar-container">
```

Comment:
Is this wrapper `div` necessary?

Reply:
In order to get the icon to show up, I needed to wrap the input with a div, and use `.searchbar-container::before` to target & add the SVG. If I used `input-container::before` the icon wouldn't appear.

Reply:
I added the div because without it I could not get the `searchIcon` or `closeIcon` to appear. I also had a tough time figuring out how to give these icons onClick events but eventually found this solution/workaround.

https://stackoverflow.com/questions/7478336/only-detect-click-event-on-pseudo-element

"This is not possible; pseudo-elements are not part of the DOM at all so you can't bind any events directly to them, you can only bind to their parent elements.

If you must have a click handler on the red region only, you have to make a child element, like a span, place it right after the opening <p> tag, apply styles to p span instead of p:before, and bind to it."

Reply:
The "pseudo-element are not part of the DOM bit" is something that I always forget. (Coincidently, this is very similar to the reason you can't dynamically style the content of `:before` and `:after` pseudo-elements like styling most other things in the DOM.)

Thank you for the explanation. Consider my comment resolved.